cli/pkg/resourcemanager/client.go

```diff
@@ -28,14 +28,23 @@ type HTTPClient struct {
 
 func NewHTTPClient(baseURL string, extraHeaders http.Header) *HTTPClient {
 	return &HTTPClient{
-		client:       http.Client{},
+		client: http.Client{
+			// this function avoids blindly followin redirects.
+			// the problem with redirects is that they don't guarantee to preserve the method, body, headers, etc.
+			// This can hide issues when developing, because the client will follow the redirect and the request
+			// will succeed, but the server will not receive the request that the user intended to send.
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		},
 		baseURL:      baseURL,
 		extraHeaders: extraHeaders,
 	}
 }
 
 func (c HTTPClient) url(resourceName string, extra ...string) *url.URL {
-	url, _ := url.Parse(fmt.Sprintf("%s/api/%s/%s", c.baseURL, resourceName, strings.Join(extra, "/")))
+	urlStr := c.baseURL + path.Join("/api", resourceName, strings.Join(extra, "/"))
```

Comment:
this changes remove a trailing `/` that we were adding to the urls without id, like `GET /transactions/` instead of `GET /transactions`. This was causing a bug. See https://github.com/kubeshop/tracetest/pull/2841/files#diff-09cf25bcafe115bcabac1fb6c58cce0a7a7b1b74536c4927f530947e659cda92R32